ls1intum/Artemis
# Fix serialization issues for metis data model

### Checklist

### Motivation and Context



This should fix flaky tests related to serialization.
![image](https://user-images.githubusercontent.com/41366522/124346636-46af8f80-dbe0-11eb-84ce-d721dd969cef.png)


### Description

I deleted the check that relied on having a course objected nested in the lecture / exercise object of a post. This test was redundant and the only reason for nesting the object post -> lecture -> course, where the flaky serialization happend.

### Steps for Testing


As always, simply use the feature and check if the code changes introduce regressions... there is no new feature added.
AS STUDENT
1. Log in to Artemis as Student
2. Navigate to Course Exercise
3. Create, edit, delete and answer posts -> save/cancel -> see if the behavior is as expected
4. Check if, in the role of a student, you are not able to edit/delete postings of others
5. Do the same for a Lecture that has posts enabled

As TUTOR
1. Log in to Artemis as Tutor
2. Navigate to Course Exercise
3. Create, edit, delete and answer posts -> save/cancel -> see if the behavior is as expected
Check if, in the role of a tutor, you are able to edit/delete postings of others
4. Check if you can approve a posting by clicking on the little check sign
5. Do the same for a Lecture that has posts enabled
